src/main/java/org/truffleruby/language/dispatch/UncachedDispatchNode.java

```diff
@@ -60,6 +61,10 @@ public Object executeDispatch(
             Object name,
             DynamicObject blockObject,
             Object[] argumentsObjects) {
+        if (RubyGuards.isForeignObject(receiverObject)) {
+            throw new UnsupportedOperationException();
+        }
```

Comment:
I've made this into an assertion, and then added a concrete check in the lookup logic.